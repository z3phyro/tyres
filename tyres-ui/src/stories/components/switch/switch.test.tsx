import { expect, test, vi } from "vitest";
import { render } from "@solidjs/testing-library";
import Switch from "./switch";

test("Switch renders", () => {
  const { getByRole } = render(() => <Switch checked={true} />);

  const switchElement = getByRole("checkbox");
  expect(switchElement).toBeTruthy();
});

test("Switch renders with label", () => {
  const { getByText } = render(() => <Switch checked={true} label="Test label" />);

  const label = getByText("Test label");
  expect(label).toBeTruthy();
});

test("Switch renders with custom class", () => {
  const { getByRole } = render(() => <Switch checked={true} class="test-class" />);

  const switchElement = getByRole("group");
  expect(switchElement).toHaveClass("test-class");
});

test("Switch calls onChange", () => {
  const onChange = vi.fn();
  const { getByRole } = render(() => <Switch checked={true} onChange={onChange} />);

  const switchElement = getByRole("checkbox");
  switchElement.click();
  expect(onChange).toHaveBeenCalled();
});

test("Switch calls onChange with correct value", () => {
  const onChange = vi.fn();
  const { getByRole } = render(() => <Switch checked={true} onChange={onChange} />);

  const switchElement = getByRole("checkbox");
  switchElement.click();
  expect(onChange).toHaveBeenCalledWith(false);
});

test("Switch calls onChange with correct value when checked is false", () => {
  const onChange = vi.fn();
  const { getByRole } = render(() => <Switch checked={false} onChange={onChange} />);

  const switchElement = getByRole("checkbox");
  switchElement.click();
  expect(onChange).toHaveBeenCalledWith(true);
});

test("Switch snapshot test", () => {
  const { asFragment } = render(() => <Switch checked={true} label="Test label" />);

  expect(asFragment()).toMatchSnapshot();
});
