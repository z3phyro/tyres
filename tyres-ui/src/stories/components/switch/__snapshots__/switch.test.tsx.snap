// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch snapshot test 1`] = `"<div role="group" id="switch-cl-12" data-checked="" class="switch"><label id="switch-cl-12-label" data-checked="" class="switch__label" for="switch-cl-12-input">Test label</label><input type="checkbox" role="checkbox" id="switch-cl-12-input" name="switch-cl-12" value="on" style="clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; border: 0px;" aria-checked="true" data-checked="" class="switch__input" aria-labelledby="switch-cl-12-label"><div data-checked="" id="switch-cl-12-control" class="switch__control"><div data-checked="" id="switch-cl-12-thumb" class="switch__thumb"></div></div></div>"`;
